### DISCUSSION TITLE
Channel update API is not publicly accessible

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#108] [General Discussions]
As part of knowlg setup, the below API is accessible only through internalIp(directly on VM). We are facing an issue while setting the knowlg infra, as other API's are accessible publicly but not the below one. 

If this API is onboarded publicly, it will help us to create the postman collection to share with any adopter to setup Knowlg. 

**Request: **

Host should be our publish domain but you can see the below curl is pointing to privateIP of the service.

`curl --location 'http://learner-service:9000/taxonomy/domain/definition' \
--header 'Content-Type: application/json' \
--header 'user-id: anil' \
--header 'Authorization: {{kong-api-token}}' \
--data '{
    "definitionNodes": [
        {
            "objectType": "Channel",
            "properties": [
                {
                    "propertyName": "name",
                    "title": "Name",
                    "description": "Name of the channel",
                    "category": "General",
                    "dataType": "Text",
                    "required": true,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{'\''inputType'\'': '\''text'\'', '\''order'\'': 1}",
                    "indexed": true
                },
                {
                    "propertyName": "description",
                    "title": "Name",
                    "description": "description of the channel",
                    "category": "General",
                    "dataType": "Text",
                    "required": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{'\''inputType'\'': '\''text'\'', '\''order'\'': 1}",
                    "indexed": true
                },
                {
                    "propertyName": "code",
                    "title": "Code",
                    "description": "Unique code for the channel",
                    "category": "General",
                    "dataType": "Text",
                    "required": true,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{'\''inputType'\'': '\''text'\'', '\''order'\'': 2}",
                    "indexed": true
                },
                {
                    "propertyName": "status",
                    "title": "Status",
                    "description": "Status of the domain",
                    "category": "general",
                    "dataType": "Select",
                    "range": [
                        "Draft",
                        "Live",
                        "Retired"
                    ],
                    "required": false,
                    "indexed": true,
                    "displayProperty": "Editable",
                    "defaultValue": "Live",
                    "renderingHints": "{'\''inputType'\'': '\''select'\'', '\''order'\'': 3}"
                },
                {
                    "propertyName": "defaultFramework",
                    "title": "Default Framework",
                    "description": "",
                    "category": "general",
                    "dataType": "Text",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Editable",
                    "defaultValue": "NCF",
                    "renderingHints": "{ '\''inputType'\'': '\''text'\'',  '\''order'\'': 10 }"
                },
                {
                    "propertyName": "createdBy",
                    "title": "Created By",
                    "description": "",
                    "category": "audit",
                    "dataType": "Text",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{ '\''inputType'\'': '\''text'\'',  '\''order'\'': 10 }"
                },
                {
                    "propertyName": "createdOn",
                    "title": "Created On",
                    "description": "",
                    "category": "audit",
                    "dataType": "Date",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Readonly",
                    "defaultValue": "",
                    "renderingHints": "{ '\''order'\'': 11 }"
                },
                {
                    "propertyName": "lastUpdatedBy",
                    "title": "Last Updated By",
                    "description": "",
                    "category": "audit",
                    "dataType": "Text",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{ '\''inputType'\'': '\''text'\'',  '\''order'\'': 12 }"
                },
                {
                    "propertyName": "lastUpdatedOn",
                    "title": "Last Updated On",
                    "description": "",
                    "category": "audit",
                    "dataType": "Date",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Readonly",
                    "defaultValue": "",
                    "renderingHints": "{ '\''order'\'': 13 }"
                },
                {
                    "propertyName": "contentFilter",
                    "title": "search",
                    "description": "",
                    "category": "technical",
                    "dataType": "json",
                    "range": [],
                    "required": false,
                    "indexed": false,
                    "displayProperty": "Readonly",
                    "defaultValue": "",
                    "renderingHints": "{ '\''order'\'': 14 }"
                },
                {
                    "propertyName": "appId",
                    "title": "appId",
                    "description": "R1.15",
                    "category": "General",
                    "dataType": "Text",
                    "required": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "",
                    "indexed": true
                },
                {
                    "propertyName": "channel",
                    "title": "Channel",
                    "description": "Channel Id of Channel",
                    "category": "General",
                    "dataType": "Text",
                    "required": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "{'\''inputType'\'': '\''text'\'', '\''order'\'': 3}",
                    "indexed": true
                },
                {
                    "propertyName": "consumerId",
                    "title": "consumerId",
                    "description": "R1.15",
                    "category": "General",
                    "dataType": "Text",
                    "required": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "",
                    "indexed": true
                },
                {
                    "propertyName": "lastStatusChangedOn",
                    "title": "lastStatusChangedOn",
                    "description": "Timestamp of last status update",
                    "category": "General",
                    "dataType": "Text",
                    "required": false,
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "",
                    "indexed": true
                },
                {
                    "required": false,
                    "dataType": "Text",
                    "propertyName": "versionKey",
                    "title": "Data Version Key",
                    "description": "Version number of Channel",
                    "category": "Technical",
                    "displayProperty": "Readonly",
                    "defaultValue": "",
                    "renderingHints": "",
                    "indexed": true,
                    "draft": false
                },
                {
                    "required": false,
                    "dataType": "Text",
                    "propertyName": "defaultLicense",
                    "title": "Default License",
                    "description": "Default License for Channel",
                    "category": "General",
                    "displayProperty": "Editable",
                    "defaultValue": "",
                    "renderingHints": "",
                    "indexed": true
                }
            ],
            "inRelations": [],
            "outRelations": [
                {
                    "relationName": "hasSequenceMember",
                    "objectTypes": [
                        "CategoryInstance"
                    ],
                    "title": "categories",
                    "description": "category instance associated with this channel",
                    "required": false,
                    "renderingHints": "{ '\''order'\'': 26 }"
                },
                {
                    "relationName": "hasSequenceMember",
                    "objectTypes": [
                        "Framework"
                    ],
                    "title": "frameworks",
                    "description": "framework associated with this channel",
                    "required": false,
                    "renderingHints": "{ '\''order'\'': 26 }"
                }
            ],
            "systemTags": [],
            "metadata": {
                "ttl": 24,
                "limit": 50
            }
        }
    ]
}'`

@Krishnaj20 @pallakartheekreddy  @gandham-santhosh

**Comment by pallakartheekreddy** [comment]
@vinukumar-vs This is expected behaviour. As per design, we are not exposed to these APIs publicly. \
@maheshkumargangula  @Krishnaj20
